wurst/systems/spells/mage/dementia/DementiaRunesNew.wurst

```diff
@@ -0,0 +1,178 @@
+
+class DementiaRune
+    DementiaRunesInstance instance
+    Runes runeType
+    //effect FX = null
+    unit dummy
+    vec3 pos
+
+    construct (DementiaRunesInstance instance, Runes runeType)
+        this.instance = instance
+        pos = instance.ownerUnit.getPos3Real()
+        this.runeType = runeType
+        if this.runeType == Runes.Lez
+            dummy = createUnit(instance.ownerUnit.getOwner(), UNIT_LOCUST_LEZ, pos, angle(0))
+        else if this.runeType == Runes.Nel
+            dummy = createUnit(instance.ownerUnit.getOwner(), UNIT_LOCUST_NEL, pos, angle(0))
+        else if this.runeType == Runes.Ka
+            dummy = createUnit(instance.ownerUnit.getOwner(), UNIT_LOCUST_KA, pos, angle(0))
+        //FX.setScale(1.5)
+        //FX.setTimeScale(0.1)
+
+    function setPos(vec3 newpos)
+        pos = newpos
+        //Adjust orbs a little lower since the SFX looks like its higher
+        //FX.setPos(pos - vec3(0,0,75))
+        dummy.setPos(pos.toVec2())
+        dummy.setFlyHeight(50, 600)
+
+    function setFacing(angle facing)
+        dummy.setFacing(facing)
+    
+    function getRandomRuneSpellID() returns DementiaRuneSpell
+        if this.runeType == Runes.Ka
+            return runeSpellsList.get(GetRandomInt(0, 1))  
+        else if this.runeType == Runes.Nel
+            return runeSpellsList.get(GetRandomInt(2, 3))
+        else
+            return runeSpellsList.get(4)
+
+
+    ondestroy
+        if instance.runes.has(this)
+            instance.runes.remove(this)
+        dummy.remove()
+        //FX.destr()
+
+public class DementiaRunesInstance
+    angle runeAngleBase
+    unit ownerUnit
+    real elapsedLifetime
+    CallbackPeriodic runeLoop = null
+    
+    LinkedList<DementiaRune> runes 
+
+    construct (unit caster)
+        ownerUnit = caster
+        elapsedLifetime = 0
+        runeAngleBase = angle(0)
+        runes = new LinkedList<DementiaRune>()
+        startRuneLoop()
+
+    function startRuneLoop()
+        runeLoop = doPeriodically(LOOP_INTERVAL) runeUpdateLoop ->
+            if ownerUnit.isAlive()
+                elapsedLifetime += LOOP_INTERVAL
+                runeAngleBase = angle(runeAngleBase.radians() + LOOP_INTERVAL * ORBIT_ANGULARSPEED * DEGTORAD)
+                updateRunePositions()
+            else
+                destroy this
+    
+    function updateRunePositions()
+        //Calculate orb angles and set it
+        angle offset = angle((360 / runes.size())*DEGTORAD)
+        int i = 0
+        vec3 ownerpos = ownerUnit.getPos3Real()
+        for r in runes
+            angle indexRuneAngle = runeAngleBase + offset * i.toReal()
+            r.setPos(ownerpos.offset2d(indexRuneAngle, ORBIT_DISTANCE))
+            r.setFacing(r.pos.angleTo2d(ownerUnit.getPos()))
+            i = i + 1
+
+    function addRune(DementiaRune rune)
+        if runes.size() < MAX_RUNE_COUNT
+            runes.add(rune)
+        else 
+            destroy runes.get(0)
```

Comment:
I created runes so that you only destroy them and dont worry about them later. The rune itself checks if its in the instances rune list and then removes itself in its onDestroy function.